index.js

```diff
@@ -58,7 +58,15 @@ module.exports.concat = function (opts, cb) {
     })
     res.on('end', function () {
       var data = Buffer.concat(chunks)
-      cb(null, res, opts.json ? JSON.parse(data) : data)
+      if (opts.json) {
+        try {
+          cb(null, res, JSON.parse(data.toString()))
+        } catch (err) {
+          cb(err, res, data)
+        }
```

Comment:
Sorry to nitpick, but there's also one other subtlety that's important to get right with try-catch statements. It's not a good idea to call the user callback in a try-catch because if their code throws an error for any reason then it will be caught here and treated like an error parsing the JSON. And their callback will get called a second time in the catch handler.

I can fix this up post-merge.